### PR TITLE
[Evals] Add endpoint to start new evaluation run

### PR DESCRIPTION
**Pull request by ssbushi**

### CODE REVIEW

File: genkit-tools/cli/src/commands/eval-flow.ts [hunk 1]
```diff
@@ -50,7 +52,10 @@ export const evalFlow = new Command('eval:flow')
   )
   .argument('<flowName>', 'Name of the flow to run')
   .argument('[data]', 'JSON data to use to start the flow')
-  .option('--input <filename>', 'JSON batch data to use to run the flow')
+  .option(
+    '--input <input>',
+    'Input dataset ID or JSON file to be used for evaluation'
```
Comment (shrutip90) [inline genkit-tools/cli/src/commands/eval-flow.ts:57]: nit: Maybe a separate --datasetid option could be more intuitive to use.

File: genkit-tools/common/src/eval/evaluate.ts [hunk 2]
```diff
@@ -35,47 +36,91 @@ import { enrichResultsWithScoring, extractMetricsMetadata } from './parser';
...
 interface BulkRunResponse {
   traceId?: string;
-  response: any;
+  response?: any;
 }
 
-/** Handles the Inference part of Inference-Evaluation cycle */
-export async function runInference(
+/**
+ * Starts a new evaluation run. Intended to be used via the reflection.
```
Comment (shrutip90) [inline genkit-tools/common/src/eval/evaluate.ts:43]: nit: reflection API?

File: genkit-tools/common/src/eval/evaluate.ts [hunk 3]
```diff
@@ -113,54 +158,72 @@ export async function getMatchingEvaluators(
...
   return responses;
 }
 
-async function fetchDataSet(
-  runner: Runner,
-  flowName: string,
-  states: BulkRunResponse[],
-  parsedData: EvalFlowInput
-): Promise<EvalInput[]> {
+async function fetchDataSet(params: {
```
Comment (shrutip90) [inline genkit-tools/common/src/eval/evaluate.ts:218]: nit: can we rename this to fetchEvalInput to distinguish from the stored datasets?

**Review by shrutip90 — APPROVED** [review]

File: genkit-tools/cli/src/commands/eval-flow.ts (hunk 1 above)
Comment (ssbushi) [reply genkit-tools/cli/src/commands/eval-flow.ts:57]: I think the versatility helps. We have to rethink the args for CLI commands anyway. We can incorporate this there. Thanks!

**Pull request merged**